src/ol/expr/gpu.js

```diff
@@ -485,6 +497,12 @@ function compile(expression, returnType, context) {
     return arrayToGlsl(/** @type {Array<number>} */ (expression.value));
   }
 
+  if ((expression.type & SizeType) > 0) {
```

Comment:
Since anything that includes the `NumberType` is already [handled above](https://github.com/openlayers/openlayers/pull/15818/files#diff-152e8cab0a3f88df63530d1530a373f48e08c10bf47096ced43afd1047020223L466), isn’t this only testing that `NumberArrayType` is included?

I’m only reviewing on a small device, so probably missing context, but it isn’t clear to me why we aren’t checking for more narrow types in all these `if`s.

Reply:
Oh, I must be missing something.  I see `NumberArrayType` is also checked right above.

Reply:
@tschaub Looks like you're not missing anything, but that code is missing tests. Looking into it now.

Reply:
I think `SizeType` should be a distinct type, like `ColorType`, instead of a broadened type. But I haven't been able to fully wrap my head around the types involved in the parsing and compilation phases to figure out how that works. So @tschaub, if you have an idea how that could be achieved, any help would be appreciated.

Reply:
Nevermind, I think I figured it out, @tschaub. Before I commit, I'm adding more tests.